extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonSerializerFactory.java

```diff
@@ -0,0 +1,416 @@
+                        return false;
+                    }
+                    ResultHandle arg = toValueReaderHandle(target, serialize, valueHandle);
+                    writeField(fieldInfo.type(), fieldName, writeFieldBranch(serialize, fieldInfo, target), jsonGenerator,
+                            serializerProvider, arg);
+                }
+            }
+        }
+        return true;
+    }
+
+    private boolean serializeMethods(ClassInfo classInfo, MethodCreator serialize, ResultHandle valueHandle,
+            ResultHandle jsonGenerator, ResultHandle serializerProvider, Set<String> serializedFields) {
+        for (MethodInfo methodInfo : classMethods(classInfo)) {
+            if (Modifier.isStatic(methodInfo.flags())) {
+                continue;
+            }
+            String fieldName = fieldNameFromMethod(methodInfo);
+            if (fieldName != null && serializedFields.add(fieldName)) {
+                if (hasUnknownAnnotation(methodInfo)) {
+                    return false;
+                }
+                ResultHandle arg = serialize.invokeVirtualMethod(MethodDescriptor.of(methodInfo), valueHandle);
+                writeField(methodInfo.returnType(), fieldName, serialize, jsonGenerator, serializerProvider, arg);
+            }
+        }
+        return true;
+    }
+
+    private void writeField(Type fieldType, String fieldName, BytecodeCreator bytecode, ResultHandle jsonGenerator,
+            ResultHandle serializerProvider, ResultHandle fieldReader) {
+        String typeName = fieldType.name().toString();
+        String primitiveMethodName = writeMethodForPrimitiveFields(typeName);
+        if (primitiveMethodName != null) {
+            MethodDescriptor primitiveWriter = MethodDescriptor.ofMethod(JSON_GEN_CLASS_NAME, primitiveMethodName, "void",
+                    "java.lang.String", typeName);
+            bytecode.invokeVirtualMethod(primitiveWriter, jsonGenerator, bytecode.load(fieldName), fieldReader);
+            return;
+        }
+
+        registerTypeToBeGenerated(fieldType, typeName);
+
+        MethodDescriptor writeMethod = MethodDescriptor.ofMethod(JSON_GEN_CLASS_NAME, "writePOJOField",
+                void.class, String.class, Object.class);
+        bytecode.invokeVirtualMethod(writeMethod, jsonGenerator, bytecode.load(fieldName), fieldReader);
+    }
+
+    private void registerTypeToBeGenerated(Type fieldType, String typeName) {
+        if (!isCollectionType(fieldType, typeName)) {
+            registerTypeToBeGenerated(typeName);
+        }
+    }
+
+    private boolean isCollectionType(Type fieldType, String typeName) {
+        if (fieldType instanceof ArrayType aType) {
+            registerTypeToBeGenerated(aType.constituent());
+            return true;
+        }
+        if (fieldType instanceof ParameterizedType pType) {
+            if (pType.arguments().size() == 1 && (typeName.equals("java.util.List") ||
```

Comment:
Adding common concrete types can be helpful here (but not key, I think - I won't expect using ArrayList to be the common case TBH)